sycl/test-e2e/syclcompat/math/math_vectorized.cpp

```diff
@@ -0,0 +1,106 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Part of the LLVM Project, under the Apache License v2.0 with LLVM
+ *  Exceptions. See https://llvm.org/LICENSE.txt for license information.
+ *  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCLcompat API
+ *
+ *  math_vectorized.cpp
+ *
+ *  Description:
+ *    math helpers for vectorized operations and fp16 operations
+ **************************************************************************/
+
+// REQUIRES: aspect-fp16
+
+// RUN: %clangxx -std=c++20 -fsycl -fsycl-targets=%{sycl_triple} %s -o %t.out
```

Comment:
Is there a reason for using c++20?

SYCL should be C++17 conformant, so we should try and not use features introduced in later C++ versions unless necessary

Reply:
This one requires a little bit of context, as you raise a good point. 
The only std++20 feature included is not part of any sycl code, used [here](https://github.com/intel/llvm/blob/46a5560c4e426ca477dc27f07e2f1eca8c72b8e3/sycl/test-e2e/syclcompat/common.hpp#L32-L38).

This is done to instantiate typed tests by using an explicit template parameter list for a lambda. The design choice came from trying to avoid adding external dependencies 

Not adding c++20, just raises a warning, so an alternative could be to switch the flag to the c++17 standard while being aware of those warnings.
If you are happy with that, I am happy to switch those back to c++17 to avoid obscuring other c++20 conformant cases that could be included by accident.

Reply:
Thanks for the explanation, I'm fine with having it be c++20, I think it is preferable to warnings